lib/init/npm-utils.js

```diff
@@ -39,6 +39,39 @@ function findPackageJson(startDir) {
     return null;
 }
 
+/**
+ *
+ * Returns the name of package manager to use,
+ * preferring yarn over npm if available
+ * @returns {string} - The package manager name
+ */
+function getPackageManager() {
+    const hasLocalYarn = fs.existsSync(path.resolve(process.cwd(), "yarn.lock"));
+    const hasLocalNpm = fs.existsSync(path.resolve(process.cwd(), "package-lock.json"));
+
+    if (hasLocalYarn) {
+        return "yarn";
+    }
+    if (hasLocalNpm) {
+        return "npm";
+    }
+    try {
+
+        /*
+         * if the sync function below fails because yarn is not installed,
+         * an error will be thrown
+         */
+        if (spawn.sync("yarn", ["--version"])) {
+            return "yarn";
+        }
+    } catch {
+
+        // Nothing
+    }
```

Comment:
just because yarn is installed does *not* mean i want to use yarn. I'm forced to have it installed for a project or two, but I have hundreds of other projects that use npm.

The default should be npm, since that's node's default.
```suggestion
```

Reply:
Done.

Reply:
Related: https://github.com/nodejs/corepack/issues/25